core/iwasm/libraries/lib-socket/test/nslookup.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <assert.h>
+#include <string.h>
+// #include <string.h>
+// #include <arpa/inet.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#else
+#include <netdb.h>
+#endif
+
+int
+main(int argc, char **argv)
+{
+    struct addrinfo hints;
+    memset(&hints, 0, sizeof(hints));
+    int v4count = 0;
+    int v6count = 0;
+
+    char *url = "google-public-dns-a.google.com";
+    struct addrinfo *res;
+    for (int af = 0; af < 2; ++af) {
```

Comment:
any specific reason for using a loop here? maybe it would be more readable to export the logic into a function and call it twice, once for ipv4 and once for ipv6

Reply:
done